### ISSUE TITLE
Loosing ability to pull from gcr when create_service_account=true is set

### ISSUE THREAD

**Issue by mnacharov** [terraform-google-modules/terraform-google-kubernetes-engine#1459]
### TL;DR

using the `create_service_account=true` flag causes http403 from the container registry (same gcp project)

### Expected behavior

`create_service_account=true` flag creates service account and adds necessary roles to it

### Observed behavior

New gke cluster with create_service_account=true parameter set unable to pull image from same project gcr 

I need to manually add `Container Registry Service Agent` role to created account - `tf-gke-<cluster>-<some-hash>@<project>.iam.gserviceaccount.com`

### Terraform Configuration

```hcl
module "gke" {
  source          = "terraform-google-modules/kubernetes-engine/google//modules/private-cluster"
  project_id      = var.project_id
  name            = var.cluster_name
  regional        = false
  region          = var.region
  zones           = slice(var.zones, 0, 1)
  release_channel = "REGULAR"

  network                 = module.gcp-network.network_name
  subnetwork              = module.gcp-network.subnets_names[0]
  ip_range_pods           = var.ip_range_pods_name
  ip_range_services       = var.ip_range_services_name
  create_service_account  = true
  enable_private_endpoint = false
  enable_private_nodes    = true
  master_ipv4_cidr_block  = "172.16.0.0/28"

  master_authorized_networks = concat([
    {
      cidr_block   = data.google_compute_subnetwork.subnetwork.ip_cidr_range
      display_name = "VPC"
    },
  ], var.master_authorized_networks)

  remove_default_node_pool = true
  node_pools = [
    {
      name               = "static"
      spot               = false
      machine_type       = "n2-standard-2"
      node_locations     = var.zones[0]
      autoscaling        = false
      node_count         = 2
      disk_type          = "pd-standard"
      image_type         = "COS_CONTAINERD"
      auto_upgrade       = true
    }
  ]
  node_pools_oauth_scopes = {
    all = [
      "https://www.googleapis.com/auth/compute",
      "https://www.googleapis.com/auth/servicecontrol",
      "https://www.googleapis.com/auth/service.management",
      "https://www.googleapis.com/auth/devstorage.read_only",
      "https://www.googleapis.com/auth/monitoring.write",
      "https://www.googleapis.com/auth/logging.write",
      "https://www.googleapis.com/auth/monitoring",
    ]
  }
}
```

### Terraform Version

```sh
Terraform v1.3.3 on darwin_arm64

  required_providers {
    google = {
      source  = "google"
      version = "~> 4.41.0"
    }
    kubernetes = {
      source = "kubernetes"
      version = "~> 2.14.0"
    }
  }
```
```


### Additional information

_No response_

**Comment by mehdicopter** [comment]
Have you tried to set `grant_registry_access` to `true` ?

**Comment by mnacharov** [comment]
No. Looks like I missed this part of documentation. Thank you!
